_posts/2025-06-25-mast-heatmap.md

```diff
@@ -0,0 +1,117 @@
+author: MAST Staff
+description: Cool MAST wallpapers
+---
+
+TL;DR - Here are the cool desktop backgrounds. Enjoy! Scroll down for the full story.
+### "Standard" Celestial Coordinates
+![mast background in "standard", surface of the Earth, coordinates](assets/img/posts/20250625/mast_wallpaper_blue.png)
+
+### Reprojection 1: Ecliptic Coordinates
+![the MAST heatmap, reprojected to the plane of the ecliptic](assets/img/posts/20250625/mast_wallpaper_ecl_blue.png)
+### Reprojection 2: Galactic Coordinates
+![the mast heatmap, projected to galactic coordinates](assets/img/posts/20250625/mast_wallpaper_gal_blue.png)
+
+## So, What am I looking at?
+You're looking at MAST's view of the sky! Specifically, each pixel in these images is colored according to the number of observations archived at those coordinates. Dark pixels have fewer observations, while brighter pixels have many.
+
+## How did you make this?
+First, consider a "typical" use of the archive: astronomers query for and access data from space telescopes. This is a two stage process: you might start by querying for HST observations of M101 in the ultraviolet range, then download all calibrated files produced by those observations. The first step is a metadata search, where you filter on telescope, wavelength, and [other fields we make searchable in our CAOM metadata database](https://mast.stsci.edu/api/v0/_c_a_o_mfields.html).
+
+MAST is not *merely* a data archive; we spend lots of time and effort making sure that our metadata are accurate, too. Given that we have over 250 million observations in our archive, this is no small feat. Using the coordinates of all observations in our database, we can create this density plot on the sky!
+## Yeah, but like, HOW specifically did you make this?
+
+### First Version: Intelligent Requests
+Our initial goal was for pixels half a degree wide, so there were `720x360=259200` regions for which we needed to know the total number of observations. Spamming our own servers with one request for each of the 2.6 million regions either requires immense patience (I'm not waiting 30 days for an answer!) or performing a [denial-of-service attack](https://en.wikipedia.org/wiki/Denial-of-service_attack) on our own servers. The other extreme would be to do a single query, loading the entire database into memory. Clearly absurd<sup>1</sup>.
+
+<small><sup>1</sup> Unless....?</small>
+
+Instead, the first successful image was produced with data from 720 [ADQL](https://en.wikipedia.org/wiki/Astronomical_Data_Query_Language) queries, avoiding a DoS attack while also working with reasonable RAM limitations. Here's roughly how the code works:
+1. Perform an ADQL query to get vertical slices of the sky. For the first slice, as an example, it would be: 
+```
+SELECT trgposRA,trgposDec FROM [REDACTED].[CaomObservation] WHERE (trgposRA < 0.5 AND targposRA >= 0)
+``` 
+Or, "Dear CAOM, please send me the coordinates of every observation with an RA between 0" and 0.5".
```

Comment:
It may be useful to clarify that `trgposRA, trgposDec` are the centroids of all observations and that this reduces potentially extended image footprints into a single point.